### ISSUE TITLE
Error: /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth: key 1 not an armored public key.

### ISSUE THREAD

**Issue by Hubbitus** [antoineco/aco-ulyaoth#4]
```
# rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
Error: /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth: key 1 not an armored public key.
```

```
# rpm -qf /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
ulyaoth-1.0.6-1.el7.centos.x86_64
```

Based on http://h30499.www3.hp.com/t5/Network-Management-OpenView-NNM/NNMi-Public-Key-installation/td-p/6306493#.VcIi_nVxJhE adding line at end of file solve problem.

**Comment by antoineco** [comment]
Thanks for the report! If I understand well the referenced commit fixes your issue? Could you tell me on what OS you're running?

**Comment by Hubbitus** [comment]
Thank you.
CentOs 7. Does it will be available as rpm? Do you have links to latest? Like ulyaoth-
latest.el7.centos.x86_64.rpm for reference them from ansible playbooks?

**Comment by antoineco** [comment]
You mean this?
https://www.ulyaoth.net/resources/ulyaoth-repository.6/

**Comment by Hubbitus** [comment]
yes

**Comment by Hubbitus** [comment]
Still version 1.0.6-1 meantime?

**Comment by antoineco** [comment]
Yes, the latest one is always available here (check the "Featured resources")
https://www.ulyaoth.net/resources/categories/repository.1/

And btw again a big thanks to @sbagmeijer for his amazing job on this.

Closing and pushing changes to the Forge

**Comment by Hubbitus** [comment]
There (kibana: https://www.ulyaoth.net/resources/ulyaoth-repository.6/) only links to https://downloads.ulyaoth.net/rpm/CentOS/x86_64/ulyaoth-1.0.6-1.el7.centos.x86_64.rpm

**Comment by eldryoth** [comment]
@Hubbitus what does not work exactly? you can install the rpm for the repository:

```
yum install https://downloads.ulyaoth.net/rpm/CentOS/x86_64/ulyaoth-1.0.6-1.el7.centos.x86_64.rpm
```

And then you can do 

```
yum install ulyaoth-kibana4
```

If you not want to install the repository (which I recommend as it will keep the package up to date) you can download Kibana here:

```
https://repos.ulyaoth.net/CentOS/7/x86_64/os/repoview/ulyaoth-kibana4.html
```

You need to manually install the gpg key then.

**Comment by Hubbitus** [comment]
I want to do not hack PGP key. Is it possible update repository packages with fix what you recently commit by that bug?

**Comment by eldryoth** [comment]
That pgp "hack" only happens if you try to install it manually because GitHub removes trailing white lines by default.

If you install the rpm there is no problem with the key:

```
[root@centos7 ~]# yum install https://downloads.ulyaoth.net/rpm/CentOS/x86_64/ulyaoth-1.0.6-1.el7.centos.x86_64.rpm
Loaded plugins: fastestmirror
ulyaoth-1.0.6-1.el7.centos.x86_64.rpm                                                                                                                                         | 6.5 kB  00:00:00
Examining /var/tmp/yum-root-8vxdd1/ulyaoth-1.0.6-1.el7.centos.x86_64.rpm: ulyaoth-1.0.6-1.el7.centos.x86_64
Marking /var/tmp/yum-root-8vxdd1/ulyaoth-1.0.6-1.el7.centos.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package ulyaoth.x86_64 0:1.0.6-1.el7.centos will be installed
--> Finished Dependency Resolution

Dependencies Resolved

=====================================================================================================================================================================================================
 Package                               Arch                                 Version                                           Repository                                                        Size
=====================================================================================================================================================================================================
Installing:
 ulyaoth                               x86_64                               1.0.6-1.el7.centos                                /ulyaoth-1.0.6-1.el7.centos.x86_64                               3.4 k

Transaction Summary
=====================================================================================================================================================================================================
Install  1 Package

Total size: 3.4 k
Installed size: 3.4 k
Is this ok [y/d/N]: y
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : ulyaoth-1.0.6-1.el7.centos.x86_64                                                                                                                                                 1/1
----------------------------------------------------------------------

Thanks for using the Ulyaoth repository!

For any additional information or help please visit my forum at:
* https://www.ulyaoth.net

----------------------------------------------------------------------
  Verifying  : ulyaoth-1.0.6-1.el7.centos.x86_64                                                                                                                                                 1/1

Installed:
  ulyaoth.x86_64 0:1.0.6-1.el7.centos

Complete!
[root@centos7 ~]# yum install ulyaoth-kibana4
Loaded plugins: fastestmirror
base                                                                                                                                                                          | 3.6 kB  00:00:00
extras                                                                                                                                                                        | 3.4 kB  00:00:00
ulyaoth                                                                                                                                                                       | 2.9 kB  00:00:00
updates                                                                                                                                                                       | 3.4 kB  00:00:00
(1/2): updates/7/x86_64/primary_db                                                                                                                                            | 3.2 MB  00:00:00
(2/2): ulyaoth/7/x86_64/primary_db                                                                                                                                            | 1.9 MB  00:00:00
Loading mirror speeds from cached hostfile
 * base: centos.uib.no
 * extras: centos.uib.no
 * updates: centos.uib.no
Resolving Dependencies
--> Running transaction check
---> Package ulyaoth-kibana4.x86_64 0:4.1.1-1.el7.centos will be installed
--> Finished Dependency Resolution

Dependencies Resolved

=====================================================================================================================================================================================================
 Package                                            Arch                                      Version                                               Repository                                  Size
=====================================================================================================================================================================================================
Installing:
 ulyaoth-kibana4                                    x86_64                                    4.1.1-1.el7.centos                                    ulyaoth                                    9.3 M

Transaction Summary
=====================================================================================================================================================================================================
Install  1 Package

Total download size: 9.3 M
Installed size: 41 M
Is this ok [y/d/N]: y
Downloading packages:
warning: /var/cache/yum/x86_64/7/ulyaoth/packages/ulyaoth-kibana4-4.1.1-1.el7.centos.x86_64.rpm: Header V4 RSA/SHA256 Signature, key ID 4665a83e: NOKEY            ] 5.6 MB/s | 6.0 MB  00:00:00 ETA
Public key for ulyaoth-kibana4-4.1.1-1.el7.centos.x86_64.rpm is not installed
ulyaoth-kibana4-4.1.1-1.el7.centos.x86_64.rpm                                                                                                                                 | 9.3 MB  00:00:02
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
Importing GPG key 0x4665A83E:
 Userid     : "Sjir Bagmeijer <sbagmeijer@ulyaoth.co.kr>"
 Fingerprint: 82db 2e02 09bd 49e4 f466 73f8 7d8b e41b 4665 a83e
 Package    : ulyaoth-1.0.6-1.el7.centos.x86_64 (@/ulyaoth-1.0.6-1.el7.centos.x86_64)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
Is this ok [y/N]: y
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : ulyaoth-kibana4-4.1.1-1.el7.centos.x86_64                                                                                                                                         1/1
----------------------------------------------------------------------

Thanks for using ulyaoth-kibana4!

Please find the official documentation for kibana here:
* http://www.elasticsearch.org/overview/kibana/

For any additional help please visit my forum at:
* http://www.ulyaoth.net

----------------------------------------------------------------------
  Verifying  : ulyaoth-kibana4-4.1.1-1.el7.centos.x86_64                                                                                                                                         1/1

Installed:
  ulyaoth-kibana4.x86_64 0:4.1.1-1.el7.centos

Complete!
```

**Comment by Hubbitus** [comment]
I have used key from rpm.

**Comment by antoineco** [comment]
That's a weird issue you're having here @Hubbitus. Anyway, I released the fix in v1.1.1 a minute ago.

**Comment by Hubbitus** [comment]
Thank you.

**Comment by Hubbitus** [comment]
By the way there https://www.ulyaoth.net/resources/ulyaoth-repository.6/ still only 1.0.6

**Comment by antoineco** [comment]
Do you need an older version?

**Comment by antoineco** [comment]
@sbagmeijer, I can confirm the issue when trying to import the key manually.

``` bash
[root@d07023197512 /]# rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth 
error: /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth: key 1 not an armored public key.
# failed

# adding newline
[root@d07023197512 /]# echo "" >> /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth

[root@d07023197512 /]# rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth 
#worked
```

I can't remember seeing this before, it might have been introduced by a rpm update.

**Comment by eldryoth** [comment]
Perhaps I not fully understand why he is manual importing the key?

if you install the rpm and then yum install a rpm from the repository it will give this warning:

```
Importing GPG key 0x4665A83E:
 Userid     : "Sjir Bagmeijer <sbagmeijer@ulyaoth.co.kr>"
 Fingerprint: 82db 2e02 09bd 49e4 f466 73f8 7d8b e41b 4665 a83e
 Package    : ulyaoth-1.0.6-1.el7.centos.x86_64 (@/ulyaoth-1.0.6-1.el7.centos.x86_64)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
Is this ok [y/N]:
```

If you then pres "y" it imports the key without any problems, I tested it on all the os systems I provide the rpm for.

**Comment by antoineco** [comment]
It's not only for manual interactions, pretty much all configuration management systems call `rpm --import` as an unattended way to import PGP keys.

As a side note, the issue doesn't happen with other PGP keys, so I assume it's due to the version you're using. But adding a newline does fix it.

**Comment by eldryoth** [comment]
Could you give this key a test and tell me if that functions correctly?:
https://repos.ulyaoth.net/RPM-GPG-KEY-ulyaoth

And how about this new spec file:
https://github.com/sbagmeijer/ulyaoth/blob/master/Repository/ulyaoth/SPECS/ulyaoth.spec

I added this:

```
%post
# Import the gpg key.
rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth

%preun
# Delete the gpg key.
rpm -e gpg-pubkey-0ea73f68-55c99ba2
```

Would that be a good way of improving the overall rpm?

**Comment by Hubbitus** [comment]
Is it common way to import keys? Is not it is exclusive user choose and action?

**Comment by eldryoth** [comment]
Yeah I did a little read more and you are right it should be up to the user to accept it or not so I changed it back to old way.

Could you tell me if the key works for you?
https://repos.ulyaoth.net/RPM-GPG-KEY-ulyaoth

**Comment by antoineco** [comment]
It does. GnuPG v1 was the culprit apparently.

**Comment by eldryoth** [comment]
Okay I am about to release a new version 1.0.7 with this new key, I am currently resigning all rpms to the new key and writing some instruction as probably some people will need to update manually.

I had a stupid mistake in the rpm file that the old key is a "config" :/ , I will make some posts on twitter so people are aware of the gpg key change.

Could you add the new key to the puppet module?
https://repos.ulyaoth.net/RPM-GPG-KEY-ulyaoth

**Comment by eldryoth** [comment]
Okay I updated the repository to 1.0.7 that should have a new fixed key, also updated all rpms to use the new key:

https://www.ulyaoth.net/resources/ulyaoth-repository.6/

I also sent you a request @antoineco with a up to date key.

**Comment by Hubbitus** [comment]
Thank yuo very much. I'll try it.

**Comment by Hubbitus** [comment]
Strait forward yum upgrade fails:

```
Downloading packages:
warning: /var/cache/yum/x86_64/7/ulyaoth/packages/ulyaoth-1.0.7-1.el7.centos.x86_64.rpm: Header V4 RSA/SHA256 Signature, key ID 0ea73f68: NOKEY
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth


The GPG keys listed for the "Ulyaoth Repository" repository are already installed but they are not correct for this package.
Check that the correct key URLs are configured for this repository.


 Failing package is: ulyaoth-1.0.7-1.el7.centos.x86_64
 GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
```

Try remove old one:
$ rpm -e gpg-pubkey-4665a83e-5516ec9a

```
Downloading packages:
warning: /var/cache/yum/x86_64/7/ulyaoth/packages/ulyaoth-1.0.7-1.el7.centos.x86_64.rpm: Header V4 RSA/SHA256 Signature, key ID 0ea73f68: NOKEY
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
Importing GPG key 0x4665A83E:
 Userid     : "Sjir Bagmeijer <sbagmeijer@ulyaoth.co.kr>"
 Fingerprint: 82db 2e02 09bd 49e4 f466 73f8 7d8b e41b 4665 a83e
 Package    : ulyaoth-1.0.6-1.el7.centos.x86_64 (@/ulyaoth-1.0.6-1.el7.centos.x86_64)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
Is this ok [y/N]: y


Public key for ulyaoth-1.0.7-1.el7.centos.x86_64.rpm is not installed


 Failing package is: ulyaoth-1.0.7-1.el7.centos.x86_64
 GPG Keys are configured as: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ulyaoth
```

**Comment by antoineco** [comment]
Because the Puppet module from the Forge does not contain the new key. Please download the module from github or wait for a new release.

**Comment by antoineco** [comment]
New key released in [v1.1.2](https://github.com/antoineco/aco-ulyaoth/releases/tag/1.1.2)
